src/OptionPicker/OptionPicker.js

```diff
@@ -8,142 +8,61 @@ import type { OptionType } from './OptionPickerTypes';
-
-  componentDidUpdate(prevProps: Props, prevState: State) {
-    const { onSelectedChange } = this.props;
-    const { selected } = this.state;
-
-    if (areSelectedOptionsChanged(selected, prevState.selected)) {
-      onSelectedChange?.(selected);
-    }
-  }
-
-  handlePressAdd = (option: OptionType) => {
-    const { onPressAdd } = this.props;
-    this.setState(
-      ({ selected }) => ({ selected: [...selected, option] }),
-      onPressAdd,
-    );
-  };
-
-  handlePressItem = (option: OptionType) => {
-    const { onPressItem } = this.props;
-    this.setState({ selected: [option] }, onPressItem);
-  };
-
-  handleClear = () => {
-    this.setState({ value: '', selected: [] });
-  };
-
-  handleChangeText = (value: string) => {
-    const { onChangeText } = this.props;
-    onChangeText(value);
-    this.setState({
-      value,
-    });
-  };
-
-  handleBackspacePress = (e: Event) => {
-    const { key } = e.nativeEvent;
-    const { value } = this.state;
-
-    if (!value && key === 'Backspace') {
-      this.removeLastSelectedOption();
-    }
-  };
-
-  removeLastSelectedOption = () => {
-    const { state } = this;
-
-    // you can remove selected place only if at least one exists
-    if (state.selected.length > 0) {
-      this.setState(({ selected }) => ({
-        selected: selected.slice(0, -1),
-      }));
-    }
-  };
-
-  render() {
-    const { selected, value } = this.state;
-    const { options, label, placeholder } = this.props;
-    const tags = selected.map(option => option.label);
-
-    return (
-      <View>
-        <View style={styles.inputWrapper}>
-          <TagsInput
-            autofocus
-            tags={tags}
-            onClearPress={this.handleClear}
-            value={value}
-            onChangeText={this.handleChangeText}
-            onKeyPress={this.handleBackspacePress}
-            label={label}
-            placeholder={placeholder}
-          />
-        </View>
-        {options && (
-          <OptionList
-            onItemPress={this.handlePressItem}
-            onAddPress={this.handlePressAdd}
-            options={filterOptions(options, selected)}
-          />
-        )}
+export default function OptionPicker(props: Props) {
+  const {
+    options,
+    label,
+    placeholder,
+    text,
+    selected,
+    onClearPress,
+    onChangeText,
+    onKeyPress,
+    onPressItem,
+    onPressAdd,
+  } = props;
+
+  const tags = selected ? selected.map(option => option.label) : [];
+
+  return (
+    <View>
+      <View style={styles.inputWrapper}>
+        <TagsInput
+          autofocus
+          tags={tags}
+          onClearPress={onClearPress}
+          value={text}
+          onChangeText={onChangeText}
+          onKeyPress={onKeyPress}
+          label={label}
+          placeholder={placeholder}
+        />
       </View>
-    );
-  }
+      {options && (
+        <OptionList
+          onItemPress={onPressItem}
+          onAddPress={onPressAdd}
+          options={options}
```

Comment:
The logic changed here for your filteredOptions.

Reply:
I know about that, I was thinking about and discussed this also with @JoseDuSV.
And the idea behind is this filtering shoul be in parent component, because sometime maby you want to have the option to choose one option more that one time. 
But I have not a strong opinion about that I can return it if you prefer to keep it here.

Reply:
Okay, let's just iterate on this. I'll accept it as it is for now because the placepicker PR has been waiting for too long, and it'd be nice to have it ready tomorrow.